components/ai_chat/core/browser/model_service.cc

```diff
@@ -119,6 +119,28 @@ const std::vector<mojom::ModelPtr>& GetLeoModels() {
       models.push_back(std::move(model));
     }
 
+    {
+      auto options = mojom::LeoModelOptions::New();
+      options->display_maker = "DeepSeek";
+      options->name = "bedrock-us.deepseek-r1-v1:0";
+      options->category = mojom::ModelCategory::CHAT;
+      options->access = mojom::ModelAccess::PREMIUM;
+      options->engine_type =
+          conversation_api ? mojom::ModelEngineType::BRAVE_CONVERSATION_API
+                           : mojom::ModelEngineType::LLAMA_REMOTE;
```

Comment:
I'm not too familiar with the engine types, but wouldn't we want to use `BRAVE_CONVERSATION_API` at all times given that we will always use Bedrock for this model? Or do we plan to use the "legacy" hosting solution as a backup?

Reply:
cc @yrliou

Reply:
@yrliou I saw that you raised https://github.com/brave/brave-browser/issues/45307. Should we keep this as is for the time being?

Reply:
The feature flag is not exposed through brave://flags, so I think it's okay to leave it as is for this PR.